docs/docs/behaviors/hold-tap.md

```diff
@@ -219,9 +237,152 @@ The following are suggested hold-tap configurations that work well with home row
 		};
 	};
 };
+```
+
+</TabItem>
+
+<TabItem value="autoshift">
+
+A popular method of implementing Autoshift in ZMK involves a C-preprocessor macro, commonly defined as `AS(keycode)`. This macro applies the `LSHIFT` modifier to the specified `keycode` when `AS(keycode)` is held, and simply performs a [keypress](key-press.md), `&kp keycode`, when the `AS(keycode)` binding is tapped. This simplifies the use of Autoshift in a keymap, as the complete hold-tap bindings for each desired Autoshift key, as in `&as LS(<keycode 1>) <keycode 1> &as LS(<keycode 2>) <keycode 2> ... &as LS(<keycode n>) <keycode n>`, can be quite cumbersome to use when applied to a large portion of the keymap.
 
+```dtsi title="Hold-Tap Example: Autoshift"
+#include <dt-bindings/zmk/keys.h>
+#include <behaviors.dtsi>
+
+#define AS(keycode) &as LS(keycode) keycode     // Autoshift Macro
+
+/ {
+    behaviors {
+        as: auto_shift {
+            compatible = "zmk,behavior-hold-tap";
+            label = "AUTO_SHIFT";
+            #binding-cells = <2>;
+            tapping_term_ms = <135>;
+            quick_tap_ms = <0>;
+            flavor = "tap-preferred";
+            bindings = <&kp>, <&kp>;
+        };
+    };
+
+	keymap {
+		compatible = "zmk,keymap";
+		default_layer {
+			bindings = <
+	            AS(Q) AS(W) AS(E) AS(R) AS(T) AS(Y) // Autoshift applied for QWERTY keys
+			>;
+		};
+	};
+};
 ```
 
-#### Comparison to QMK
+</TabItem>
+
+<TabItem value="sticky_holds">
+
+This section describes hold-tap behaviors involving [sticky keys](sticky-key.md) or [sticky layers](sticky-layer.md). These code snippets implement a sticky tap paired with a [keypress](key-press.md) or [momentary layer](layers.md/#momentary-layer) on the hold, as relevant to each example.
+
+If uninterrupted by another keypress, a regular [sticky key](sticky-key.md) or [sticky layer](sticky-layer.md) remains sticky after its release; the [`release-after-ms`](sticky-key.md/#release-after-ms) timer must expire before the sticky key/layer is deactivated. This can be an issue for cases like using the mouse while a modifier is held. The "sticky hold" implementations below provide a workaround for this problem, since the user can immediately deactivate the [sticky key](sticky-key.md) or [sticky layer](sticky-layer.md) on its release, after it has been held down for a sufficient amount of time.
+
+```dtsi title="Hold-Tap Example: Sticky Hold Key (&sk on tap, &kp on hold)"
+#include <dt-bindings/zmk/keys.h>
+#include <behaviors.dtsi>
+
+#define SKH(keycode) &skh keycode keycode   // Macro to apply sticky-key-on-tap/keypress-on-hold to a specific keycode
```

Comment:
Typo: this should be be `SHK` instead of `SKH` and `&shk` instead of `&skh` to be consistent with code below
```suggestion
#define SHK(keycode) &shk keycode keycode   // Macro to apply sticky-key-on-tap/keypress-on-hold to a specific keycode
```